### PR TITLE
[PL/CT] MPL for component transport process

### PR DESCRIPTION
**Pull request by renchao-lu**
- Material properties involved in the component transport process all invoked from MPL.
- Linear property extended for linear dependency on multiple variables.

1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.2.2)
2. [x] Tests covering your feature were added?
3. [ ] Any new feature or behavior change was documented?

### CODE REVIEW

File: ProcessLib/ComponentTransport/ComponentTransportProcessData.h [hunk 1]
```diff
@@ -53,9 +46,7 @@ struct ComponentTransportProcessData
     {
     }
 
-    MaterialLib::PorousMedium::PorousMediaProperties porous_media_properties;
     ParameterLib::Parameter<double> const& fluid_reference_density;
```
Comment (endJunction) [inline ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: The fluid reference density seems to be also unused.

Comment (renchao-lu) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: I think so too. Shall I ?

File: MaterialLib/MPL/CreateProperty.cpp [hunk 2]
```diff
@@ -179,7 +186,7 @@ std::unique_ptr<MaterialPropertyLib::Property> createProperty(
             //! \ogs_file_param{properties__property__Parameter__parameter_name}
             config.getConfigParameter<std::string>("parameter_name");
         auto const& parameter = ParameterLib::findParameter<double>(
-            parameter_name, parameters, 1, nullptr);
+            parameter_name, parameters, 0, nullptr);
```
Comment (TomFischer) [inline MaterialLib/MPL/CreateProperty.cpp]: What is the reason of this change?

File: ProcessLib/ComponentTransport/ComponentTransportProcessData.h [hunk 3]
```diff
@@ -32,19 +31,13 @@ namespace ComponentTransport
...
             media_map_,
         ParameterLib::Parameter<double> const& decay_rate_,
         Eigen::VectorXd const& specific_body_force_, bool const has_gravity_,
         bool const non_advective_form_)
-        : porous_media_properties(std::move(porous_media_properties_)),
-          fluid_reference_density(fluid_reference_density_),
-          fluid_properties(std::move(fluid_properties_)),
+        : fluid_reference_density(fluid_reference_density_),
           media_map(std::move(media_map_)),
           decay_rate(decay_rate_),
```
Comment (TomFischer) [inline ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: Shouldn't the decay rate also be handled as MPL property?

File: Tests/Data/Elliptic/square_100x100_ComponentTransport/square_1e4_heterogeneity.prj [hunk 4]
```diff
@@ -67,9 +41,36 @@
...
+                    <name>permeability</name>
+                    <type>Parameter</type>
+                    <parameter_name>kappa1</parameter_name>
+                </property>
+                <property>
+                    <name>porosity</name>
+                    <type>Parameter</type>
+                    <parameter_name>constant_porosity_parameter</parameter_name>
+                </property>
+                <property>
```
Comment (TomFischer) [inline Tests/Data/Elliptic/square_100x100_ComponentTransport/square_1e4_heterogeneity.prj]: Is the storage used anywhere? If not you could remove this tags from all project files.

**Comment by TomFischer** [comment]
For discussion:
In this PR the permeability is handled  as a property of the media, not as a property of the solid phase. In contrast to this in the HT process the permeability is a solid phase property. The approach in this PR is probably physical more correct?! To keep it consistent should I change the HT process also?

File: MaterialLib/MPL/CreateProperty.cpp [hunk 5]
```diff
@@ -115,30 +115,37 @@ std::unique_ptr<MaterialPropertyLib::Property> createProperty(
             //! \ogs_file_param{properties__property__LinearProperty__reference_value}
             config.getConfigParameter<double>("reference_value");
 
-        auto const& independent_variable_config =
+        std::vector<MaterialPropertyLib::IndependentVariable> ivs;
+        auto const& independent_variables_config =
             //! \ogs_file_param{properties__property__LinearProperty__independent_variable}
```
Comment (TomFischer) [inline MaterialLib/MPL/CreateProperty.cpp]: The documentation path has to be updated.
```suggestion
            //! \ogs_file_param{properties__property__LinearProperty__independent_variables}
```

File: MaterialLib/MPL/CreateProperty.cpp [hunk 6]
```diff
@@ -115,30 +115,37 @@ std::unique_ptr<MaterialPropertyLib::Property> createProperty(
...
-
-        MaterialPropertyLib::IndependentVariable const iv{
-            ivt, reference_condition, slope};
+            config.getConfigSubtree("independent_variables");
+        for (auto const& independent_variable_config :
+             independent_variables_config.getConfigSubtreeList(
+                 "independent_variable"))
+        {
+            auto const& variable_name =
+                //! \ogs_file_param{properties__property__LinearProperty__independent_variable__variable_name}
```
Comment (TomFischer) [inline MaterialLib/MPL/CreateProperty.cpp]:
```suggestion
                //! \ogs_file_param{properties__property__LinearProperty__independent_variables__independent_variable__variable_name}
```

File: MaterialLib/MPL/CreateProperty.cpp [hunk 7]
```diff
@@ -115,30 +115,37 @@ std::unique_ptr<MaterialPropertyLib::Property> createProperty(
...
+                independent_variable_config.getConfigParameter<std::string>(
+                    "variable_name");
+            auto const reference_condition =
+                //! \ogs_file_param{properties__property__LinearProperty__independent_variable__reference_condition}
```
Comment (TomFischer) [inline MaterialLib/MPL/CreateProperty.cpp]:
```suggestion
                //! \ogs_file_param{properties__property__LinearProperty__independent_variables__independent_variable__reference_condition}
```

File: MaterialLib/MPL/CreateProperty.cpp [hunk 8]
```diff
@@ -115,30 +115,37 @@ std::unique_ptr<MaterialPropertyLib::Property> createProperty(
...
+                independent_variable_config.getConfigParameter<double>(
+                    "reference_condition");
+            auto const slope =
+                //! \ogs_file_param{properties__property__LinearProperty__independent_variable__slope}
```
Comment (TomFischer) [inline MaterialLib/MPL/CreateProperty.cpp]:
```suggestion
                //! \ogs_file_param{properties__property__LinearProperty__independent_variables__independent_variable__slope}
```

**Comment by TomFischer** [comment]
Project file tag documentation has to be updated.

File: MaterialLib/MPL/CreateProperty.cpp (hunk 2 above)
Comment (renchao-lu) [reply MaterialLib/MPL/CreateProperty.cpp]: The parameter found in the subtree `<parameters>/<parameter>` can be read anyway, irrespective of number of components.

File: ProcessLib/ComponentTransport/ComponentTransportProcessData.h (hunk 3 above)
Comment (renchao-lu) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: Technically viable. I am not sure it proper to consider decay rate as part of material physical properties.

File: MaterialLib/MPL/CreateProperty.cpp (hunk 8 above)
Comment (renchao-lu) [reply MaterialLib/MPL/CreateProperty.cpp]: Fixed.

File: MaterialLib/MPL/CreateProperty.cpp (hunk 7 above)
Comment (renchao-lu) [reply MaterialLib/MPL/CreateProperty.cpp]: Fixed.

File: MaterialLib/MPL/CreateProperty.cpp (hunk 5 above)
Comment (renchao-lu) [reply MaterialLib/MPL/CreateProperty.cpp]: Fixed.

File: MaterialLib/MPL/CreateProperty.cpp (hunk 6 above)
Comment (renchao-lu) [reply MaterialLib/MPL/CreateProperty.cpp]: Fixed.

File: ProcessLib/ComponentTransport/ComponentTransportProcessData.h (hunk 3 above)
Comment (montoyav) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: In my opinion decay should not be consider as material property as this property is specific of radioisotopes or what we consider "components".

**Comment by renchao-lu** [comment]
> Project file tag documentation has to be updated.

Updated.

File: MaterialLib/MPL/CreateProperty.cpp (hunk 2 above)
Comment (TomFischer) [reply MaterialLib/MPL/CreateProperty.cpp]: @renchao-lu That is true. So the change isn't necessary?

Comment (endJunction) [reply MaterialLib/MPL/CreateProperty.cpp]: @TomFischer 0 means, that the number of components is arbitrary.

File: ProcessLib/ComponentTransport/ComponentTransportProcessData.h (hunk 3 above)
Comment (endJunction) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: If I remember correctly, the decay rate was used for tracers too.

Comment (endJunction) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: Whether the decay rate goes into MPL or not, depends on where and how it is used. I didn't look into the details, so no suggestions for now.

Comment (TomFischer) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: When the decay rate is a property of a component we should add it as a MPL component .

File: Tests/Data/Elliptic/square_100x100_ComponentTransport/square_1e4_heterogeneity.prj (hunk 4 above)
Comment (renchao-lu) [reply Tests/Data/Elliptic/square_100x100_ComponentTransport/square_1e4_heterogeneity.prj]: Dropped.

Remark: Just retain one in the project file of Theis benchmark. In the project file, it is documented how to consider storage in an indirect way.

File: ProcessLib/ComponentTransport/ComponentTransportProcessData.h (hunk 1 above)
Comment (renchao-lu) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: Removed.

File: ProcessLib/ComponentTransport/ComponentTransportProcessData.h (hunk 3 above)
Comment (renchao-lu) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: Now it turns clear to me. Decay rate should be under the subtree of aqueous liquid phase and assigned as component property.

Comment (montoyav) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: > If I remember correctly, the decay rate was used for tracers too.

If tracers are not considered components, then decay it is also relevant for tracers.

Comment (renchao-lu) [reply ProcessLib/ComponentTransport/ComponentTransportProcessData.h]: Decay rate invoked from MPL.

**Comment by renchao-lu** [comment]
> For discussion:
> In this PR the permeability is handled as a property of the media, not as a property of the solid phase. In contrast to this in the HT process the permeability is a solid phase property. The approach in this PR is probably physical more correct?! To keep it consistent should I change the HT process also?

i think so.
